clients/spark-extensions-base/src/main/scala/org/apache/spark/sql/execution/datasources/v2/NessieUtils.scala

```diff
@@ -229,6 +230,12 @@ object NessieUtils {
       .getAllWithPrefix(s"spark.sql.catalog.$catalogName.")
       .toMap
       .asJava
+
+    // When org.apache.spark.sql.DataFrameReader.load() is called with tableName.
+    // Catalog is created using SQLConf.
+    // So, update the default reference name in SQLConf with the current reference.
+    SQLConf.mergeSparkConf(SQLConf.get, SparkSession.active.sparkContext.conf)
```

Comment:
Why does it work with this call? I am not sure whether it makes a lot of sense to blindly update all properties.

Reply:
`SQLConf` will have `nessie.ref = "main"`, active session will have `nessie.ref = $current_branch` from USE REFERENCE command.

 `org.apache.spark.sql.DataFrameReader.load()` is building the nessie catalog using `SQLConf`. Hence, it always points to the original default branch. So, compaction throws table not found error as it looks in the wrong branch (as mentioned in the issue description)